src/pages/index.js

```diff
@@ -48,52 +78,116 @@ const cardListEl = new Section(
 cardListEl.renderItems();
 
 const userInfo = new UserInfo({
-  nameSelector: ".profile__title",
-  descriptionSelector: ".profile__description",
+  title: ".profile__title",
+  description: ".profile__description",
+  avatarSelector: ".profile__image",
 });
 
+api
+  .getUserInfo()
+  .then((userData) => {
+    userInfo.setUserInfo({
+      title: userData.name,
+      description: userData.about,
+      avatar: userData.avatar,
+    });
+  })
+  .catch((err) => console.error(err));
+
 const cardImageModal = new PopupWithImage("#preview-image-modal");
 cardImageModal.setEventListeners();
 
 function handleImageClick(cardData) {
   cardImageModal.open(cardData);
 }
 
-const addCardPopup = new PopupWithForm(
-  "#add-card-modal",
-  handleAddCardFormSubmit
+const deleteModalConfirmation = new PopupConfirmation(
+  "#modal-confirm",
+  async (cardId, cardElement) => {
+    try {
+      await api.deleteCard(cardId);
+      cardElement.remove();
+    } catch (err) {
+      console.error(`Error On Card Deletion ${err}`);
+    }
+  }
 );
-addCardPopup.setEventListeners();
+
+function handleDeleteCard(card) {
+  deleteModalConfirmation.setSubmitAction(() => {
+    api
+      .deleteCard(card.getId())
+      .then(() => {
+        card.remove();
+        deleteModalConfirmation.close();
+      })
+      .catch((err) => {
+        console.error(err);
+      });
+  });
+  deleteModalConfirmation.open();
+}
+
+function handleLikeClick(card) {
+  if (card.setIsLiked) {
+    api.dislikeCard(card._id).then(() => {});
```

Comment:
* You need to **toggle likes** only in a `then` block after a successful response from the server because if there is an error, the user will not know about that and will see the toggled heart color.


```js

function handleLikeClick(card) {
  if (card.setIsLiked) {
    api
      .dislikeCard(card._id)
      .then(() => {
        card.handleLikeIcon();  // here
      })
      .catch(console.error);
  } else {
```